test/governance/ATMTokenTest.js

```diff
@@ -0,0 +1,190 @@
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _4_mint_vesting_basic: [daoMember2, 1000, 7000, undefined, false],
+        _5_mint_vesting_above_cap: [daoMember2, 21000, 7000, 'ERC20Capped: cap exceeded', true],
+        _6_mint_vesting_zero_address: [NULL_ADDRESS, 3000, 60000, "ERC20: mint to the zero address", true],
+        _7_mint_vesting_incorrect_address: [daoMember1, 1000, 7000, 'Account does not have a vesting balance!', true],
+    },function(
+        receipent,
+        amount,
+        vestingPeriod,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to mint correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                let result;
+                await instance.mintVesting(receipent, amount, vestingPeriod, { from: daoAgent });
+                if (receipent == daoMember1) {
+                    result = await instance.isVested(daoMember2);
+                } else {
+                    result = await instance.isVested(receipent);
+                }
+                // Assertions
+                assert(!mustFail, 'It should have failed because the amount is greater than the cap');
+                assert(result);
+            } catch (error) {
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _8_set_supply_cap_basic: [70000, daoAgent, undefined, false],
+        _9_set_supply_cap_invalid_sender: [100000, daoMember1, 'PauserRole: caller does not have the Pauser role', true]
+    },function(
+        newCap,
+        sender,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to set cap correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                await instance.setCap(newCap, { from: sender });
+                const result = await instance.cap();
+                // Assertions
+                assert(!mustFail, 'It should have failed because the sender is invalid');
+                assert(result);
+            } catch (error) {
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _10_claim_vested_basic: [daoMember2, 1000, 7000, 7001, undefined, false],
```

Comment:
We usually use a counter per function we are testing (for the scenario), instead of a counter for all scenarios.
Also, if the Test JS file starts to grow, please consider creating a file per function, it means:

ATMTokenMintTest.js
ATMTokenMintVestingTest.js
ATMTokenWithdrawVestedTest.js
etc

Reply:
Understood, split the tests